flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/Levels.java

```diff
@@ -43,6 +43,16 @@ public class Levels {
 
     public Levels(Comparator<RowData> keyComparator, List<DataFileMeta> inputFiles, int numLevels) {
         this.keyComparator = keyComparator;
+
+        // in case the num of levels is not specified explicitly
+        numLevels =
+                Math.max(
+                        numLevels,
+                        inputFiles.stream()
```

Comment:
NIT: We can extract a field here: `int restoreMaxLevel = ...`. The format of the java stream in the method parameters doesn't look very good.